tree/dataframe/CMakeLists.txt

```diff
@@ -159,10 +159,6 @@ if(MSVC)
   target_compile_definitions(ROOTDataFrame PRIVATE _USE_MATH_DEFINES)
 endif()
 
-if(builtin_nlohmannjson)
-  target_include_directories(ROOTDataFrame PRIVATE ${CMAKE_SOURCE_DIR}/builtins)
-else()
-  target_link_libraries(ROOTDataFrame PUBLIC nlohmann_json::nlohmann_json)
-endif()
+target_link_libraries(ROOTDataFrame PUBLIC nlohmann_json::nlohmann_json)
```

Comment:
The same pattern with the `if(builtin_nlohmannjson)` also appears in `roofit/jsoninterface/CMakeLists.txt`. Can we also replace it there?

Reply:
Sure, but you need to disable it if json is off. Does RF handle the case correctly where no json is found?

Reply:
Thanks! This failure on mac12 of [projectroot.roofit.hs3.test.gtest_roofit_hs3_test_testHS3HistFactory](https://lcgapp-services.cern.ch/root-jenkins/job/root-pullrequests-build/168507/testReport/projectroot.roofit.hs3/test/gtest_roofit_hs3_test_testHS3HistFactory/) scares me a bit by the way. If the RooFit code is so fragile that changing the build system makes it crash it's not good :( I wonder why it only appeared once, maybe a non-deterministic destruction order problem that is unrelated to the changes in this PR...

Reply:
@guitargeek Check the two new commits. There were even more instances that were added after this PR had been opened.

Reply:
> Thanks! This failure on mac12 of [projectroot.roofit.hs3.test.gtest_roofit_hs3_test_testHS3HistFactory](https://lcgapp-services.cern.ch/root-jenkins/job/root-pullrequests-build/168507/testReport/projectroot.roofit.hs3/test/gtest_roofit_hs3_test_testHS3HistFactory/) scares me a bit by the way. If the RooFit code is so fragile that changing the build system makes it crash it's not good :( I wonder why it only appeared once, maybe a non-deterministic destruction order problem that is unrelated to the changes in this PR...

That would scare me, too. I don't think it's related to the build system. Check this with asan maybe?

Reply:
Yes, good idea :+1: